### PR TITLE
Test several feature weights for V2 Ad Serving

### PR DESCRIPTION
**Pull request by moritzhaller**
Resolves https://github.com/brave/brave-variations/issues/181

Following #123 we want to test different combinations of feature parameters for V2 ad serving. The griffin parameter under test is a string of 7 floats, decomposing into the following feature weights:

1. DoesMatchIntentChildSegments
2. DoesMatchIntentParentSegments
3. DoesMatchInterestChildSegments
4. DoesMatchInterestParentSegments
5. AdLastSeenHoursAgo
6. AdvertiserLastSeenHoursAgo
7. Priority

Thus we aim to split the study into 5 groups as follows:

1. (20%) Ad Serving V1
2. (20%) Ad Serving V2 with default weights (all weights are set to 1.0)
3. (20%) Ad Serving V2 with "16.0, 8.0, 16.0, 8.0, 0.0, 0.0, 0.0" (isolate segment feature effects)
4. (20%) Ad Serving V2 with "0.0, 0.0, 0.0, 0.0, 16.0, 16.0, 0.0" (isolate recency feature effects)
5. (20%) Ad Serving V2 with "32.0, 16.0, 16.0, 8.0, 4.0, 2.0, 1.0" (somewhat arbitrary weights w/ bias towards segment features)
6. In addition to Nightly, the study should run in Beta channel.

### CODE REVIEW

**Comment by moritzhaller** [comment]
verified study takes effect with two runs of 

yielding

```
[4618:259:1215/095944.888374:VERBOSE1:study_filtering.cc(372)] Kept study BraveAds.AdServingStudy.
```

Run 1:
```
BraveAds.AdServingStudy:MaximumAdNotificationsPerDay=100/MaximumInlineContentAdsPerHour=8/MaximumInlineContentAdsPerDay=40/AdServingVersion=1
```

Run 2:
```
BraveAds.AdServingStudy:MaximumAdNotificationsPerDay=100/MaximumInlineContentAdsPerHour=8/MaximumInlineContentAdsPerDay=40/AdServingVersion=2/
```

**Review by kjozwiak — APPROVED** [review]
Uplift into `production` approved after deliberating with @brave/uplift-approvers.

**Pull request merged**
